### PR TITLE
Add submit multiple jobs

### PR DESCRIPTION
**Pull request by fviale**
- SchedulerRestInterface: multipleSubmitFromUrls
 - Scheduler: submit list of jobs
 - SchedulerClient: multipleSubmitFromUrl
 etc

 Additional parts of this commit:
 - improved SchedulingMethodImpl performance using factorization of some operations that can be done once instead of for each task to be launched. Use caches when possible
 - improved SubmitHandler concurrency by ensuring that frontendState is notified as soon as the job is started

### CODE REVIEW

File: rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/common/SchedulerRestInterface.java [hunk 1]
```diff
@@ -1564,6 +1552,22 @@ JobIdData submitFromUrl(@HeaderParam("sessionid") String sessionId, @HeaderParam
...
+     * @param jsonBody
+     *            a list of workflows. Each element must contain an url, a set of variables and generic information
+     * @return the list of objects containing, for each workflow a <code>jobid</code> if submission was successful, or the error description if an error occurred.
+     */
+    @POST
+    @Path("{path:jobs}/body/multi")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    List<JobIdDataAndError> multipleSubmitFromUrls(@HeaderParam("sessionid") String sessionId,
+            List<WorkflowUrlData> jsonBody) throws NotConnectedRestException;
```
Comment (nebilbenmabrouk) [inline rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/common/SchedulerRestInterface.java:1569]: @fviale would you please give an example of the JSON body of the request. Thanks in advance

**Review by nebilbenmabrouk — APPROVED** [review]
LGTM

**Review by marounkoussaifi — APPROVED** [review]

File: rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/common/SchedulerRestInterface.java (hunk 1 above)
Comment (fviale) [reply rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/common/SchedulerRestInterface.java:1569]: Input:
```
[
  {
    variables: {variables values set at submission ...
      string =>string
    }
    genericInformation: {generic information values set at submission ...
      string =>string
    }
    workflowUrl:stringUrl of the workflow that will be fetched ...
  }
]
```
OUTPUT:
```
[
  {
    stackTrace:string
    id:number
    error:boolean
    errorMessage:string
    readableName:string
    fetchError:boolean
  }
]
```

**Review by AlexFalessi — APPROVED** [review]
=== Please ignore ==
Meant to approve another PR

**Pull request merged**